code/lab1-tdd-triangle/src/main/java/com/lahtachev/triangle/Triangle.java

```diff
@@ -0,0 +1,118 @@
+package com.lahtachev.triangle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Math.*;
+
+public class Triangle {
+    private final Point point1;
+    private final Point point2;
+    private final Point point3;
+    private final int sideNumber = 3;
+
+    public Triangle(final Point point1, final Point point2, final Point point3) {
+        if (point1.equals(point2) || point2.equals(point3)) {
+            throw new IllegalArgumentException("Points should not be in one place");
+        }
+        if (point3.getX() - point1.getX() * (point2.getY() - point1.getY())
+                == (point3.getY() - point1.getY()) * (point2.getX() - point1.getX())) {
+            throw new IllegalArgumentException("Points should not be on one line");
+        }
+        this.point1 = point1;
+        this.point2 = point2;
+        this.point3 = point3;
+    }
+
+    public Point getPoint1() {
+        return point1;
+    }
+
+    public Point getPoint2() {
+        return point2;
+    }
+
+    public Point getPoint3() {
+        return point3;
+    }
+
+    public double getSide(final Point point1, final Point point2) {
+        return sqrt(pow(point1.getX() - point2.getX(), 2) + pow(point1.getY() - point2.getY(), 2));
+    }
+
+    public double getAngle(final Point anglePoint,
+                           final Point sidePoint1,
+                           final Point sidePoint2) {
+        double angle = acos(((sidePoint1.getX() - anglePoint.getX())
```

Comment:
I am also confused because in some methods, for example, this one, the user can get getAngle points that do not belong to the triangle. Maybe you should do this method private and take only all angles, Bisectors, Medians, Heights?

Reply:
Done